R/qenv-eval_code.R

```diff
@@ -1,16 +1,21 @@
 #' Evaluate code in `qenv`
 #'
 #' @details
+#'
 #' `eval_code()` evaluates given code in the `qenv` environment and appends it to the `code` slot.
 #' Thus, if the `qenv` had been instantiated empty, contents of the environment are always a result of the stored code.
+#' The `qenv` object is immutable, even though it inherits from the environment class, which is typically mutable by
+#' design. This means that each code evaluation does not modify the original `qenv` object directly. Instead, every
+#' code evaluation creates a new `qenv` object that reflects the result of the changes, leaving the original object
+#' unchanged.
 #'
 #' @param object (`qenv`)
```

Comment:
:+1: to using refer `qenv` as "environment" instead of "object"

As a side note, I was thinking if we should deprecate this argument name in favor of something else:
-  `envir` is too generic as we  don't accept base environments
- I would like `qenv` but it might be too specific
- `data` would make it similar to `within` arguments, but not very descriptive

Keeping as is feels the best approach

<\side note>

Reply:
Crazy idea that would allow `eval_code` to accept base environments and convert them to qenv, but I cannot justify it as:

1. We shouldn't support the same in `within` as that's a base-level generic/object pair
2. It also doesn't make sense in terms of using it
    - Just as it doesn't make sense calling `eval_code`/`within` after `teal_data(iris = iris, mtcars = mtcars)`
        - Although in this case it's possible.

Reply:
> As a side note, I was thinking if we should deprecate this argument name in favor of something else:

I'd add a `x` to this list also

> Crazy idea that would allow eval_code to accept base environments and convert them to qenv, but I cannot justify it as

I like where it goes but at this moment we can be pretty confident with the design. However, replacing `teal_data(a = a, b = b)` constructor with an environment which could be supported by `eval_code` sounds also nice.

> Just as it doesn't make sense calling eval_code/within after teal_data(iris = iris, mtcars = mtcars)

It makes some sense. You can initialize `teal_data` object with some code and data as `!verified` and it could still be a reproducible code and data.

Reply:
> You can initialize teal_data object with some code and data as !verified and it could still be a reproducible code and data.

You got a point here, I see people doing this for computational heavy operations that would be avoided, or if the code/values come from a different "system"